src/opensynth/models/faraday/gaussian_mixture/fit_gmm.py

```diff
@@ -130,6 +124,7 @@ def fit_gmm(
         vae_module,
         num_components,
         num_features,
+        num_datapoints=data.dataset.n_samples,
```

Comment:
Is this the number of samples in the training set, or the total dataset? I think we want it to be the number of samples in the training set, but I am not 100% sure ...

Reply:
In the example notebook we define the data module like so:

`gmm_data_module = LCLDataModule(data_path=data_path, stats_path=stats_path, batch_size=5000, n_samples=50000, outlier_path=outlier_path)`

We then train the model with this data module: `faraday_model_10.train_gmm(dm=gmm_data_module)`

The `n_samples` set in the `gmm_data_module` is what will be used as the number of datapoints when fitting the gmm on that data module. I think this is the desired behaviour?

Reply:
Yep looks good, thanks for checking!